### PR TITLE
feat: show license status in identity UI

### LINKED ISSUE #20739: C8 License - Show C8 License status in Identity
## Related to

Epic: https://github.com/camunda/product-hub/issues/2359

Dependent On: https://github.com/camunda/camunda/issues/20540

## Overview

Once the License endpoint in the monorepo is available, the front end needs to consume data from the endpoint and display it to the user. Use the `C3Navigation` component to display the status. The label will only show in Self Managed, it will not show in SaaS

The endpoint: 
```
GET /license
{
   "validLicense": true,
   "licenseType": "self-managed"
}
```
`validLicense` will be `true` or `false`. The value is determined whether the Camunda license is valid or not
`licenseType` will return one of three strings, `saas`, `self-managed`, or `unknown`. The value reflects the current running mode Camunda is in. `unknown` is returned when a license is invalid or missing

The C3 Navbar component contains a feature which can easily display the license status. [See here for the prop to use](https://camunda-composite-components.pages.dev/?path=/docs/uishell-c3navigation--docs#navbar-prop). The license prop is available in `@camunda/camunda-composite-components` version `0.7.0` and higher

The value of `validLicense` maps directly to the `isProductionLicense` prop for the header. `show` will be `true` when `environment` is equal to `self-managed`

Console Self-Managed has implemented a license status already: https://github.com/camunda-cloud/camunda-cloud-management-apps/blob/ff518c29429d34363e9fd678b66da7762cadb7d2/apps/console-sm-frontend/src/components/navbar.tsx#L104

## Actions

- Consume the `camunda-license` endpoint and read the license status. The status is then displayed on the front end for the user

```[tasklist]
### Pull Requests
- [ ] tbd
```

## Example Screenshot with Navbar props

Screenshot 2024-08-07 at 7 25 17 AM

### PR DESCRIPTION
**Pull request by maryarm**
## Description
Once the License endpoint in the monorepo is available, the front end needs to consume data from the endpoint and display it to the user. Use the `C3Navigation` component to display the status. The label will only show in Self Managed, it will not show in SaaS

The endpoint: 
```
GET /license
{
   "validLicense": true,
   "licenseType": "self-managed"
}
```
`validLicense` will be `true` or `false`. The value is determined whether the Camunda license is valid or not
`licenseType` will return one of three strings, `saas`, `self-managed`, or `unknown`. The value reflects the current running mode Camunda is in. `unknown` is returned when a license is invalid or missing.

## Related issues

closes #20739

### CODE REVIEW

File: identity/client/src/components/layout/AppHeader.tsx [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 
 const AppHeader = () => {
   const routes = useGlobalRoutes();
   const navigate = useNavigate();
+  const { data: license } = useApi(checkLicense);
+  const showLicense: boolean =
+    license == null
+      ? false
+      : license.licenseType !== undefined &&
+        license.licenseType == "self-managed";
```
Comment (matthewBearCamunda) [inline identity/client/src/components/layout/AppHeader.tsx:15]: `showLicense` should be looking for `saas`, that's the only case where it the license would be hidden. Otherwise, it would show

So, a small change is needed here

```
const showLicense: boolean =
    license == null
      ? false
      : license.licenseType !== undefined &&
        license.licenseType != "saas";
```

File: identity/client/src/pages/users/List.tsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ const List: FC = () => {
     <Page>
       <EntityList
         title={t("Users")}
-        data={userSearchResults!.items}
+        data={userSearchResults == null ? [] : userSearchResults.items}
```
Comment (matthewBearCamunda) [inline identity/client/src/pages/users/List.tsx:74]: Is this change needed for the license call to work?

**Review by matthewBearCamunda — CHANGES_REQUESTED** [review]
One small change and one question!

File: identity/client/src/pages/users/List.tsx (hunk 2 above)
Comment (maryarm) [reply identity/client/src/pages/users/List.tsx:74]: It's not directly related as it fixes another bug I discovered during the test, but without this the page can't be rendered and just shows error page.

File: identity/client/src/components/layout/AppHeader.tsx [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+import { checkLicense } from "src/utility/api/headers";
 
 const AppHeader = () => {
   const routes = useGlobalRoutes();
   const navigate = useNavigate();
+  const { data: license } = useApi(checkLicense);
+  const showLicense: boolean =
+    license == null
+      ? true
+      : license.licenseType === undefined && license.licenseType != "saas";
```
Comment (matthewBearCamunda) [inline identity/client/src/components/layout/AppHeader.tsx:14]: This should have a `||` operator rather than `&&`

**Review by matthewBearCamunda — CHANGES_REQUESTED** [review]

**Review by matthewBearCamunda — APPROVED** [review]
LGTM
